lib/ansible/modules/network/nxos/nxos_bgp_neighbor.py

```diff
@@ -307,9 +350,23 @@ def state_present(module, existing, proposed, candidate):
     commands = list()
     proposed_commands = apply_key_map(PARAM_TO_COMMAND_KEYMAP, proposed)
     existing_commands = apply_key_map(PARAM_TO_COMMAND_KEYMAP, existing)
-
+    attribute_string = ''
     for key, value in proposed_commands.items():
-        if value is True:
+        if key == "local-as":
+            # If local_as arg is passed to module execution -- build command string
+            if module.params['local_as'] is not None:
+                local_as_attributes = ['local_as_no_prepend', 'local_as_replace_as', 'local_as_dual_as']
+                attribute_string = ' '.join([PARAM_TO_DEFAULT_KEYMAP.get(x) for x in local_as_attributes if module.params[x] is True])
+                command = 'local-as {0} {1}'.format(module.params['local_as'], attribute_string).strip()
+                # this will loop for each local_as_attribute -- only append one time.
+                if command not in commands:
+                    commands.append(command)
+            # If local_as arg is NOT passed to module execution -- only remove if it existed originally
+            elif module.params['local_as'] is None and existing.get('local_as'):
+                command = 'no local-as {0}'.format(existing.get('local_as'))
```

Comment:
@trishnaguha Trishna, I reviewed this code; LGTM except for this if block which removes optional (non-playbook) values from the device. My understanding from Mike is that (for existing modules) we should be merging playbook changes into existing device configs. What's your take on this?
Thanks

Reply:
Agreed.  I don't think we want the default behavior to remove existing configuration from the device.  If we end up supporting `overridden` state in the future then this would be a supported workflow.

Reply:
Thanks for the review! I removed this elif-block and modified the appropriate unit test to ensure that the default behaviour does not override existing configuration on the device.